homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -68,9 +68,27 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the FRITZ!SmartHome binary sensor from ConfigEntry."""
-    coordinator: FritzboxDataUpdateCoordinator = hass.data[FRITZBOX_DOMAIN][
-        entry.entry_id
-    ][CONF_COORDINATOR]
+    coordinator: FritzboxDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        CONF_COORDINATOR
+    ]
+
+    async def _add_new_devices(event: Event) -> None:
+        """Add newly discovered devices."""
+        async_add_entities(
+            [
+                FritzboxBinarySensor(coordinator, ain, description)
+                for ain, device in coordinator.data.devices.items()
+                if ain in event.data.get("ains", [])
+                for description in BINARY_SENSOR_TYPES
+                if description.suitable(device)
+            ]
```

Comment:
All this code is duplicated from the original call to add entities. Better to have a common function for both.

Reply:
There is always the additional ckeck `if ain in event.data.get("ains", [])` included, to just add new devices. So it needs a common function, which can differentiate between "add all" or "add only the new one" 🤔

Reply:
They will always be new the first time around. Just have the set of added devices always available